sdks/java/core/src/main/java/org/apache/beam/sdk/io/range/ByteKey.java

```diff
@@ -43,10 +45,10 @@
   public static final ByteKey EMPTY = ByteKey.of();
 
   /**
-   * Creates a new {@link ByteKey} backed by the specified {@link ByteString}.
+   * Creates a new {@link ByteKey} backed by a copy of the data in the specified {@link ByteBuffer}.
```

Comment:
Creates a new {@link ByteKey} by copying the remainder of the data in the specified {@link ByteBuffer}.

remainder is key here, which means the person can use methods to set the position and mark the end to have the bytekey represent a subrange.